### PR TITLE
[test_macsec] Add testcase interop with protocols

### PR DESCRIPTION
**Pull request by jimmyzhai**
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

Please provide following information to help code review process a bit easier:
-->
### Description of PR
<!--
- Please include a summary of the change and which issue is fixed.
- Please also include relevant motivation and context. Where should reviewer start? background context?
- List any dependencies that are required for this change.
-->

Summary:
1/ Add neighbor cEOS support 
2/ Add Testcase TestInteropProtocol

### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)


### Back port request
- [ ] 201911
- [ ] 202012

### Approach
#### What is the motivation for this PR?
Implement testcase macsec interop with protocols in [MACsec testplan](https://github.com/judyjoseph/sonic-mgmt/blob/macsec_tests/docs/testplan/MACsec-test-plan.md#testcase--macsec-interop-with-other-slow-protocols).

#### How did you do it?
Add macsec test with lacp, lldp, bgp and snmp

#### How did you verify/test it?

Verify it in DUT `str2-a7280cr3-4` with t0 topo:
```
junhuazhai@e25918d5b081:/var/src/sonic-mgmt-int/tests$ ./run_tests.sh -u -n vms28-t0-7280-4 -d str2-a7280cr3-4 -c macsec/test_macsec.py -f ../ansible/testbed.yaml -i ../ansible/str2,../ansible/veos -e "--skip_sanity --disable_loganalyzer -vvl" 
=== Running tests in groups ===
/usr/local/lib/python2.7/dist-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography.exceptions import InvalidSignature
===================================================================================================================== test session starts ======================================================================================================================
platform linux2 -- Python 2.7.17, pytest-4.6.5, py-1.11.0, pluggy-0.13.1 -- /usr/bin/python
cachedir: .pytest_cache
metadata: {'Python': '2.7.17', 'Platform': 'Linux-4.10.0-42-generic-x86_64-with-Ubuntu-18.04-bionic', 'Packages': {'py': '1.11.0', 'pytest': '4.6.5', 'pluggy': '0.13.1'}, 'Plugins': {u'repeat': u'0.9.1', u'ansible': u'2.2.2', u'xdist': u'1.28.0', u'allure-pytest': u'2.8.22', u'html': u'1.22.1', u'forked': u'1.3.0', u'metadata': u'1.11.0'}}
ansible: 2.8.12
rootdir: /var/src/sonic-mgmt-int/tests, inifile: pytest.ini
plugins: forked-1.3.0, metadata-1.11.0, xdist-1.28.0, html-1.22.1, repeat-0.9.1, allure-pytest-2.8.22, ansible-2.2.2
collecting ... /usr/local/lib/python2.7/dist-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography.exceptions import InvalidSignature
collected 96 items                                                                                                                                                                                                                                             

macsec/test_macsec.py::TestControlPlane::test_wpa_supplicant_processes[GCM-AES-128-security-true] PASSED                                                                                                                                                 [  1%]
macsec/test_macsec.py::TestControlPlane::test_appl_db[security-GCM-AES-128-true] PASSED                                                                                                                                                                  [  2%]
macsec/test_macsec.py::TestControlPlane::test_mka_session[security-GCM-AES-128-true] PASSED                                                                                                                                                              [  3%]
macsec/test_macsec.py::TestDataPlane::test_server_to_neighbor[GCM-AES-128-security-true] SKIPPED                                                                                                                                                         [  4%]
macsec/test_macsec.py::TestDataPlane::test_dut_to_neighbor[GCM-AES-128-security-true] PASSED                                                                                                                                                             [  5%]
macsec/test_macsec.py::TestDataPlane::test_neighbor_to_neighbor[GCM-AES-128-security-true] SKIPPED                                                                                                                                                       [  6%]
macsec/test_macsec.py::TestFaultHandling::test_link_flap[GCM-AES-128-security-true] PASSED                                                                                                                                                               [  7%]
macsec/test_macsec.py::TestFaultHandling::test_mismatch_macsec_configuration[GCM-AES-128-security-true] PASSED                                                                                                                                           [  8%]
macsec/test_macsec.py::TestInteropProtocol::test_port_channel[GCM-AES-128-security-true] PASSED                                                                                                                                                          [  9%]
macsec/test_macsec.py::TestInteropProtocol::test_lldp[GCM-AES-128-security-true] PASSED                                                                                                                                                                  [ 10%]
macsec/test_macsec.py::TestInteropProtocol::test_bgp[GCM-AES-128-security-true] PASSED                                                                                                                                                                   [ 11%]
macsec/test_macsec.py::TestInteropProtocol::test_snmp[GCM-AES-128-security-true] PASSED                                                                                                                                                                  [ 12%]
macsec/test_macsec.py::TestControlPlane::test_wpa_supplicant_processes[GCM-AES-128-security-false] SKIPPED                                                                                                                                               [ 13%]
macsec/test_macsec.py::TestControlPlane::test_appl_db[security-GCM-AES-128-false] SKIPPED                                                                                                                                                                [ 14%]
macsec/test_macsec.py::TestControlPlane::test_mka_session[security-GCM-AES-128-false] SKIPPED                                                                                                                                                            [ 15%]
macsec/test_macsec.py::TestDataPlane::test_server_to_neighbor[GCM-AES-128-security-false] SKIPPED                                                                                                                                                        [ 16%]
macsec/test_macsec.py::TestDataPlane::test_dut_to_neighbor[GCM-AES-128-security-false] SKIPPED                                                                                                                                                           [ 17%]
macsec/test_macsec.py::TestDataPlane::test_neighbor_to_neighbor[GCM-AES-128-security-false] SKIPPED                                                                                                                                                      [ 18%]
macsec/test_macsec.py::TestFaultHandling::test_link_flap[GCM-AES-128-security-false] SKIPPED                                                                                                                                                             [ 19%]
macsec/test_macsec.py::TestFaultHandling::test_mismatch_macsec_configuration[GCM-AES-128-security-false] SKIPPED                                                                                                                                         [ 20%]
macsec/test_macsec.py::TestInteropProtocol::test_port_channel[GCM-AES-128-security-false] SKIPPED                                                                                                                                                        [ 21%]
macsec/test_macsec.py::TestInteropProtocol::test_lldp[GCM-AES-128-security-false] SKIPPED                                                                                                                                                                [ 22%]
macsec/test_macsec.py::TestInteropProtocol::test_bgp[GCM-AES-128-security-false] SKIPPED                                                                                                                                                                 [ 23%]
macsec/test_macsec.py::TestInteropProtocol::test_snmp[GCM-AES-128-security-false] SKIPPED                                                                                                                                                                [ 25%]
macsec/test_macsec.py::TestControlPlane::test_wpa_supplicant_processes[GCM-AES-256-security-false] SKIPPED                                                                                                                                               [ 26%]
macsec/test_macsec.py::TestControlPlane::test_appl_db[security-GCM-AES-256-false] SKIPPED                                                                                                                                                                [ 27%]
macsec/test_macsec.py::TestControlPlane::test_mka_session[security-GCM-AES-256-false] SKIPPED                                                                                                                                                            [ 28%]
macsec/test_macsec.py::TestDataPlane::test_server_to_neighbor[GCM-AES-256-security-false] SKIPPED                                                                                                                                                        [ 29%]
macsec/test_macsec.py::TestDataPlane::test_dut_to_neighbor[GCM-AES-256-security-false] SKIPPED                                                                                                                                                           [ 30%]
macsec/test_macsec.py::TestDataPlane::test_neighbor_to_neighbor[GCM-AES-256-security-false] SKIPPED                                                                                                                                                      [ 31%]
macsec/test_macsec.py::TestFaultHandling::test_link_flap[GCM-AES-256-security-false] SKIPPED                                                                                                                                                             [ 32%]
macsec/test_macsec.py::TestFaultHandling::test_mismatch_macsec_configuration[GCM-AES-256-security-false] SKIPPED                                                                                                                                         [ 33%]
macsec/test_macsec.py::TestInteropProtocol::test_port_channel[GCM-AES-256-security-false] SKIPPED                                                                                                                                                        [ 34%]
macsec/test_macsec.py::TestInteropProtocol::test_lldp[GCM-AES-256-security-false] SKIPPED                                                                                                                                                                [ 35%]
macsec/test_macsec.py::TestInteropProtocol::test_bgp[GCM-AES-256-security-false] SKIPPED                                                                                                                                                                 [ 36%]
macsec/test_macsec.py::TestInteropProtocol::test_snmp[GCM-AES-256-security-false] SKIPPED                                                                                                                                                                [ 37%]
macsec/test_macsec.py::TestControlPlane::test_wpa_supplicant_processes[GCM-AES-256-security-true] PASSED                                                                                                                                                 [ 38%]
macsec/test_macsec.py::TestControlPlane::test_appl_db[security-GCM-AES-256-true] PASSED                                                                                                                                                                  [ 39%]
macsec/test_macsec.py::TestControlPlane::test_mka_session[security-GCM-AES-256-true] PASSED                                                                                                                                                              [ 40%]
macsec/test_macsec.py::TestDataPlane::test_server_to_neighbor[GCM-AES-256-security-true] SKIPPED                                                                                                                                                         [ 41%]
macsec/test_macsec.py::TestDataPlane::test_dut_to_neighbor[GCM-AES-256-security-true] PASSED                                                                                                                                                             [ 42%]
macsec/test_macsec.py::TestDataPlane::test_neighbor_to_neighbor[GCM-AES-256-security-true] SKIPPED                                                                                                                                                       [ 43%]
macsec/test_macsec.py::TestFaultHandling::test_link_flap[GCM-AES-256-security-true] PASSED                                                                                                                                                               [ 44%]
macsec/test_macsec.py::TestFaultHandling::test_mismatch_macsec_configuration[GCM-AES-256-security-true] PASSED                                                                                                                                           [ 45%]
macsec/test_macsec.py::TestInteropProtocol::test_port_channel[GCM-AES-256-security-true] PASSED                                                                                                                                                          [ 46%]
macsec/test_macsec.py::TestInteropProtocol::test_lldp[GCM-AES-256-security-true] PASSED                                                                                                                                                                  [ 47%]
macsec/test_macsec.py::TestInteropProtocol::test_bgp[GCM-AES-256-security-true] PASSED                                                                                                                                                                   [ 48%]
macsec/test_macsec.py::TestInteropProtocol::test_snmp[GCM-AES-256-security-true] PASSED                                                                                                                                                                  [ 50%]
macsec/test_macsec.py::TestControlPlane::test_wpa_supplicant_processes[GCM-AES-XPN-128-security-false] SKIPPED                                                                                                                                           [ 51%]
macsec/test_macsec.py::TestControlPlane::test_appl_db[security-GCM-AES-XPN-128-false] SKIPPED                                                                                                                                                            [ 52%]
macsec/test_macsec.py::TestControlPlane::test_mka_session[security-GCM-AES-XPN-128-false] SKIPPED                                                                                                                                                        [ 53%]
macsec/test_macsec.py::TestDataPlane::test_server_to_neighbor[GCM-AES-XPN-128-security-false] SKIPPED                                                                                                                                                    [ 54%]
macsec/test_macsec.py::TestDataPlane::test_dut_to_neighbor[GCM-AES-XPN-128-security-false] SKIPPED                                                                                                                                                       [ 55%]
macsec/test_macsec.py::TestDataPlane::test_neighbor_to_neighbor[GCM-AES-XPN-128-security-false] SKIPPED                                                                                                                                                  [ 56%]
macsec/test_macsec.py::TestFaultHandling::test_link_flap[GCM-AES-XPN-128-security-false] SKIPPED                                                                                                                                                         [ 57%]
macsec/test_macsec.py::TestFaultHandling::test_mismatch_macsec_configuration[GCM-AES-XPN-128-security-false] SKIPPED                                                                                                                                     [ 58%]
macsec/test_macsec.py::TestInteropProtocol::test_port_channel[GCM-AES-XPN-128-security-false] SKIPPED                                                                                                                                                    [ 59%]
macsec/test_macsec.py::TestInteropProtocol::test_lldp[GCM-AES-XPN-128-security-false] SKIPPED                                                                                                                                                            [ 60%]
macsec/test_macsec.py::TestInteropProtocol::test_bgp[GCM-AES-XPN-128-security-false] SKIPPED                                                                                                                                                             [ 61%]
macsec/test_macsec.py::TestInteropProtocol::test_snmp[GCM-AES-XPN-128-security-false] SKIPPED                                                                                                                                                            [ 62%]
macsec/test_macsec.py::TestControlPlane::test_wpa_supplicant_processes[GCM-AES-XPN-128-security-true] SKIPPED                                                                                                                                            [ 63%]
macsec/test_macsec.py::TestControlPlane::test_appl_db[security-GCM-AES-XPN-128-true] SKIPPED                                                                                                                                                             [ 64%]
macsec/test_macsec.py::TestControlPlane::test_mka_session[security-GCM-AES-XPN-128-true] SKIPPED                                                                                                                                                         [ 65%]
macsec/test_macsec.py::TestDataPlane::test_server_to_neighbor[GCM-AES-XPN-128-security-true] SKIPPED                                                                                                                                                     [ 66%]
macsec/test_macsec.py::TestDataPlane::test_dut_to_neighbor[GCM-AES-XPN-128-security-true] SKIPPED                                                                                                                                                        [ 67%]
macsec/test_macsec.py::TestDataPlane::test_neighbor_to_neighbor[GCM-AES-XPN-128-security-true] SKIPPED                                                                                                                                                   [ 68%]
macsec/test_macsec.py::TestFaultHandling::test_link_flap[GCM-AES-XPN-128-security-true] SKIPPED                                                                                                                                                          [ 69%]
macsec/test_macsec.py::TestFaultHandling::test_mismatch_macsec_configuration[GCM-AES-XPN-128-security-true] SKIPPED                                                                                                                                      [ 70%]
macsec/test_macsec.py::TestInteropProtocol::test_port_channel[GCM-AES-XPN-128-security-true] SKIPPED                                                                                                                                                     [ 71%]
macsec/test_macsec.py::TestInteropProtocol::test_lldp[GCM-AES-XPN-128-security-true] SKIPPED                                                                                                                                                             [ 72%]
macsec/test_macsec.py::TestInteropProtocol::test_bgp[GCM-AES-XPN-128-security-true] SKIPPED                                                                                                                                                              [ 73%]
macsec/test_macsec.py::TestInteropProtocol::test_snmp[GCM-AES-XPN-128-security-true] SKIPPED                                                                                                                                                             [ 75%]
macsec/test_macsec.py::TestControlPlane::test_wpa_supplicant_processes[GCM-AES-XPN-256-security-false] SKIPPED                                                                                                                                           [ 76%]
macsec/test_macsec.py::TestControlPlane::test_appl_db[security-GCM-AES-XPN-256-false] SKIPPED                                                                                                                                                            [ 77%]
macsec/test_macsec.py::TestControlPlane::test_mka_session[security-GCM-AES-XPN-256-false] SKIPPED                                                                                                                                                        [ 78%]
macsec/test_macsec.py::TestDataPlane::test_server_to_neighbor[GCM-AES-XPN-256-security-false] SKIPPED                                                                                                                                                    [ 79%]
macsec/test_macsec.py::TestDataPlane::test_dut_to_neighbor[GCM-AES-XPN-256-security-false] SKIPPED                                                                                                                                                       [ 80%]
macsec/test_macsec.py::TestDataPlane::test_neighbor_to_neighbor[GCM-AES-XPN-256-security-false] SKIPPED                                                                                                                                                  [ 81%]
macsec/test_macsec.py::TestFaultHandling::test_link_flap[GCM-AES-XPN-256-security-false] SKIPPED                                                                                                                                                         [ 82%]
macsec/test_macsec.py::TestFaultHandling::test_mismatch_macsec_configuration[GCM-AES-XPN-256-security-false] SKIPPED                                                                                                                                     [ 83%]
macsec/test_macsec.py::TestInteropProtocol::test_port_channel[GCM-AES-XPN-256-security-false] SKIPPED                                                                                                                                                    [ 84%]
macsec/test_macsec.py::TestInteropProtocol::test_lldp[GCM-AES-XPN-256-security-false] SKIPPED                                                                                                                                                            [ 85%]
macsec/test_macsec.py::TestInteropProtocol::test_bgp[GCM-AES-XPN-256-security-false] SKIPPED                                                                                                                                                             [ 86%]
macsec/test_macsec.py::TestInteropProtocol::test_snmp[GCM-AES-XPN-256-security-false] SKIPPED                                                                                                                                                            [ 87%]
macsec/test_macsec.py::TestControlPlane::test_wpa_supplicant_processes[GCM-AES-XPN-256-security-true] SKIPPED                                                                                                                                            [ 88%]
macsec/test_macsec.py::TestControlPlane::test_appl_db[security-GCM-AES-XPN-256-true] SKIPPED                                                                                                                                                             [ 89%]
macsec/test_macsec.py::TestControlPlane::test_mka_session[security-GCM-AES-XPN-256-true] SKIPPED                                                                                                                                                         [ 90%]
macsec/test_macsec.py::TestDataPlane::test_server_to_neighbor[GCM-AES-XPN-256-security-true] SKIPPED                                                                                                                                                     [ 91%]
macsec/test_macsec.py::TestDataPlane::test_dut_to_neighbor[GCM-AES-XPN-256-security-true] SKIPPED                                                                                                                                                        [ 92%]
macsec/test_macsec.py::TestDataPlane::test_neighbor_to_neighbor[GCM-AES-XPN-256-security-true] SKIPPED                                                                                                                                                   [ 93%]
macsec/test_macsec.py::TestFaultHandling::test_link_flap[GCM-AES-XPN-256-security-true] SKIPPED                                                                                                                                                          [ 94%]
macsec/test_macsec.py::TestFaultHandling::test_mismatch_macsec_configuration[GCM-AES-XPN-256-security-true] SKIPPED                                                                                                                                      [ 95%]
macsec/test_macsec.py::TestInteropProtocol::test_port_channel[GCM-AES-XPN-256-security-true] SKIPPED                                                                                                                                                     [ 96%]
macsec/test_macsec.py::TestInteropProtocol::test_lldp[GCM-AES-XPN-256-security-true] SKIPPED                                                                                                                                                             [ 97%]
macsec/test_macsec.py::TestInteropProtocol::test_bgp[GCM-AES-XPN-256-security-true] SKIPPED                                                                                                                                                              [ 98%]
macsec/test_macsec.py::TestInteropProtocol::test_snmp[GCM-AES-XPN-256-security-true] SKIPPED                                                                                                                                                             [100%]

------------------------------------------------------------------------------------------------ generated xml file: /var/src/sonic-mgmt-int/tests/logs/tr.xml -------------------------------------------------------------------------------------------------
=================================================================================================================== short test summary info ====================================================================================================================
SKIPPED [24] /var/src/sonic-mgmt-int/tests/macsec/conftest.py:51: GCM-AES-XPN-128 is not supported on neighbor EOS
SKIPPED [24] /var/src/sonic-mgmt-int/tests/macsec/conftest.py:51: GCM-AES-XPN-256 is not supported on neighbor EOS
SKIPPED [24] /var/src/sonic-mgmt-int/tests/macsec/conftest.py:83: EOS with send_sci false does not work due to portchannel mac not matching ether port mac!
SKIPPED [4] /var/src/sonic-mgmt-int/tests/common/plugins/ptfadapter/__init__.py:180: Neighbor devices aren't SONiC so that the ptf nn service cannot be started
=========================================================================================================== 20 passed, 76 skipped in 1852.72 seconds ===========================================================================================================
```

#### Any platform specific information?
vSonic, Arista 7280

#### Supported testbed topology if it's a new test case?

### Documentation
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

### CODE REVIEW

**Comment by judyjoseph** [comment]
@jimmyzhai please could you add the test run logs to the PR desription as well

File: tests/macsec/test_macsec.py [hunk 1]
```diff
@@ -205,3 +245,122 @@ def check_mka_establishment():
         disable_macsec_port(duthost, port_name)
         disable_macsec_port(nbr["host"], nbr["port"])
         delete_macsec_profile(nbr["host"], profile_name)
+
+def get_lldp_list(host):
```
Comment (Pterosaur) [inline tests/macsec/test_macsec.py:249]: This function should be moved to macsec_platform_helper.py

File: tests/macsec/test_macsec.py [hunk 2]
```diff
@@ -205,3 +245,122 @@ def check_mka_establishment():
...
+        '''
+        LLDP_ADVERTISEMENT_INTERVAL = 30 # default interval in seconds
+        LLDP_HOLD_MULTIPLIER = 4 # default multiplier number
+        LLDP_TIMEOUT = LLDP_ADVERTISEMENT_INTERVAL * LLDP_HOLD_MULTIPLIER
+
+        # select one macsec link
+        ctrl_port, nbr = ctrl_links.items()[0]
+        # TODO: vsonic vm has issue on lldp
+        if not isinstance(nbr["host"], EosHost):
+            return
```
Comment (Pterosaur) [inline tests/macsec/test_macsec.py:301]: Use pytest.skip

**Review by Pterosaur — CHANGES_REQUESTED** [review]

File: tests/macsec/conftest.py [hunk 3]
```diff
@@ -52,8 +44,8 @@ def profile_name():
 def default_priority():
     return 64
 
-
-@pytest.fixture(scope="module", params=["GCM-AES-128", "GCM-AES-256", "GCM-AES-XPN-128", "GCM-AES-XPN-256"])
+#@pytest.fixture(scope="module", params=["GCM-AES-128", "GCM-AES-256", "GCM-AES-XPN-128", "GCM-AES-XPN-256"])
+@pytest.fixture(scope="module", params=["GCM-AES-128"])
 def cipher_suite(request):
```
Comment (Pterosaur) [inline tests/macsec/conftest.py:49]: pytest.skip

File: tests/macsec/macsec_helper.py [hunk 4]
```diff
@@ -14,7 +14,6 @@
 from macsec_common_helper import *
 from macsec_platform_helper import *
 
-
```
[the remark this answers is not in the document]
Comment (Pterosaur) [reply tests/macsec/macsec_helper.py:17]: Alt+Shift+F format

File: tests/macsec/test_macsec.py [hunk 5]
```diff
@@ -79,10 +82,27 @@ def _test_mka_session():
...
+        for ctrl_port in ctrl_links.keys():
+            up_link = upstream_links[ctrl_port]
+            dut_macaddress = duthost.get_dut_iface_mac(ctrl_port)
+            payload = "{} -> {}".format(down_link["name"], up_link["name"])
+            logging.info(payload)
+            # Source mac address is not useful in this test case and we use an arbitrary mac address as the source
+            pkt = create_pkt(
+                "00:01:02:03:04:05", dut_macaddress, "1.2.3.4", up_link["ipv4_addr"], bytes(payload))
+            exp_pkt = create_exp_pkt(pkt, pkt[scapy.IP].ttl - 1)
+            testutils.send_packet(
```
Comment (Pterosaur) [inline tests/macsec/test_macsec.py:99]: Remove. Update it in multiple members of portchannel

File: tests/macsec/test_macsec.py [hunk 6]
```diff
@@ -79,10 +82,27 @@ def _test_mka_session():
...
+                ptfadapter, down_link["ptf_port_id"], pkt, TestDataPlane.BATCH_COUNT)
+            check_macsec_pkt(macsec_attr=get_macsec_attr(duthost, ctrl_port), test=ptfadapter,
+                             ptf_port_id=up_link["ptf_port_id"],  exp_pkt=exp_pkt, timeout=10)
```
Comment (Pterosaur) [inline tests/macsec/test_macsec.py:102]: Remove. Update it in multiple members of portchannel

File: tests/macsec/conftest.py [hunk 7]
```diff
@@ -16,31 +16,23 @@ def pytest_configure(config):
...
         for nbr in [n["host"] for n in nbrhosts.values()] + [duthost]:
+            if isinstance(nbr, EosHost):
+                continue
             if len(nbr.shell("docker ps | grep macsec | grep -v grep")["stdout_lines"]) != 1:
                 return False
             if len(nbr.shell("ps -ef | grep macsecmgrd | grep -v grep")["stdout_lines"]) != 1:
                 return False
         return True
-    assert wait_until(180, 1, 1, check_macsec_enabled)
+    assert wait_until(180, 5, 5, check_macsec_enabled)
```
[the remark this answers is not in the document]
Comment (jimmyzhai) [reply tests/macsec/conftest.py:32]: docker macsec mostly takes much time to start up. Use a bit large delay and interval time, avoiding frequent retry/failure.

File: tests/macsec/macsec_helper.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jimmyzhai) [reply tests/macsec/macsec_helper.py:17]: Fix it

File: tests/macsec/test_macsec.py (hunk 1 above)
Comment (jimmyzhai) [reply tests/macsec/test_macsec.py:249]: Fix it

File: tests/macsec/conftest.py (hunk 3 above)
Comment (jimmyzhai) [reply tests/macsec/conftest.py:49]: Fix it

File: tests/macsec/test_macsec.py (hunk 6 above)
Comment (jimmyzhai) [reply tests/macsec/test_macsec.py:102]: Ok. Do it commonly while multiple members of portchannel for both vsonic and ceos neighbors.

**Review by Pterosaur — APPROVED** [review]

**Review by Pterosaur — COMMENTED** [review]
Wrong Approve click

File: tests/macsec/macsec_platform_helper.py [hunk 8]
```diff
@@ -122,3 +129,25 @@ def find_portchannel_from_member(port_name, portchannel_list):
         if port_name in v["members"]:
             return v
     return None
+
```
Comment (Pterosaur) [inline tests/macsec/macsec_platform_helper.py:132]: Here should be two empty lines for PEP8 standard

File: tests/macsec/test_macsec.py [hunk 9]
```diff
@@ -205,3 +229,109 @@ def check_mka_establishment():
...
+        '''
+        LLDP_ADVERTISEMENT_INTERVAL = 30  # default interval in seconds
+        LLDP_HOLD_MULTIPLIER = 4  # default multiplier number
+        LLDP_TIMEOUT = LLDP_ADVERTISEMENT_INTERVAL * LLDP_HOLD_MULTIPLIER
+
+        # select one macsec link
+        for ctrl_port, nbr in ctrl_links.items():
+            # TODO: vsonic vm has issue on lldp
+            if not isinstance(nbr["host"], EosHost):
+                pytest.skip("test_lldp has issue with vsonic neighbor")
```
Comment (Pterosaur) [inline tests/macsec/test_macsec.py:269]: Please add 
`| jq '.DEVICE_METADATA.localhost.mac |= sub("(?<=^\w)[26aeAE]"; "c")'`
after 
https://github.com/Azure/sonic-mgmt/blob/3cc50dcc7d25ebc4ee33baafaf3f009bc32aa175/ansible/roles/sonic/tasks/vsonic.yml#L27
, I believe that could fix this issue in vsonic
image

**Review by Pterosaur — APPROVED** [review]

**Pull request merged**